File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/dependencyinjection/AndroidViewModule.java
package com.example.sampleapplicationforgm.dependencyinjection;

import com.example.sampleapplicationforgm.ui.DetailActivity;
import com.example.sampleapplicationforgm.ui.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class AndroidViewModule {

    @ContributesAndroidInjector
    abstract MainActivity bindMainActivity();

    @ContributesAndroidInjector
    abstract DetailActivity bindDetailActivity();
}

File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/dependencyinjection/AndroidViewModule.kt
package com.example.sampleapplicationforgm.dependencyinjection

import com.example.sampleapplicationforgm.ui.DetailActivity
import com.example.sampleapplicationforgm.ui.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class AndroidViewModule {
    @ContributesAndroidInjector
    abstract fun bindMainActivity(): MainActivity?

    @ContributesAndroidInjector
    abstract fun bindDetailActivity(): DetailActivity?
}